linera-execution/src/runtime.rs

```diff
@@ -39,7 +40,7 @@ pub struct SyncRuntimeInternal<const WRITABLE: bool> {
     execution_state_sender: ExecutionStateSender,
 
     /// The current stack of application descriptions.
-    applications: Vec<ApplicationStatus>,
+    applications: LinkedHashMap<UserApplicationId, ApplicationStatus>,
```

Comment:
This feels a bit overkill. I would just use keep the `Vec` and add a separate `HashSet<UserApplicationId>`.

Reply:
One disadvantage of having two types is that they must always be updated together. Forgetting to do so can lead to bugs :/

Reply:
Thanks!